src/components/QRShare/QRShareWithDownload/index.js

```diff
@@ -1,43 +1,53 @@
-import React, {Component} from 'react';
-import {withNetwork} from '@components/OnyxProvider';
+import React, {forwardRef, useImperativeHandle, useRef} from 'react';
 import getQrCodeFileName from '@components/QRShare/getQrCodeDownloadFileName';
 import {qrShareDefaultProps, qrSharePropTypes} from '@components/QRShare/propTypes';
+import useNetwork from '@hooks/useNetwork';
 import fileDownload from '@libs/fileDownload';
 import QRShare from '..';
 
-class QRShareWithDownload extends Component {
-    qrShareRef = React.createRef();
-
-    constructor(props) {
-        super(props);
-
-        this.download = this.download.bind(this);
-    }
-
-    download() {
-        return new Promise((resolve, reject) => {
-            // eslint-disable-next-line es/no-optional-chaining
-            const svg = this.qrShareRef.current?.getSvg();
-            if (svg == null) {
-                return reject();
-            }
-
-            svg.toDataURL((dataURL) => resolve(fileDownload(dataURL, getQrCodeFileName(this.props.title))));
-        });
-    }
-
-    render() {
-        return (
-            <QRShare
-                ref={this.qrShareRef}
-                // eslint-disable-next-line react/jsx-props-no-spreading
-                {...this.props}
-                logo={this.props.network.isOffline ? null : this.props.logo}
-            />
-        );
-    }
+function QRShareWithDownload({innerRef, ...props}) {
```

Comment:
I guess you can pass `innerRef` as a second param, and then just put `QRShareWithDownload` in the `forwardRef` (see comment below)
```suggestion
function QRShareWithDownload(props, innerRef) {
```

Reply:
I tried this initially, but as we are still using propTypes and defaultProps I get a warning from React saying that they won't worked when used directly with `forwardRef`.